autorest/responder.go

```diff
@@ -135,3 +142,22 @@ func WithErrorUnlessStatusCode(codes ...int) RespondDecorator {
 func WithErrorUnlessOK() RespondDecorator {
 	return WithErrorUnlessStatusCode(http.StatusOK)
 }
+
+// ExtractHeader extracts all values of the specified header from the http.Response. It returns an
+// empty string slice if the passed http.Response is nil or the header does not exist.
+func ExtractHeader(header string, resp *http.Response) []string {
```

Comment:
you can potentially return a nil slice as well, semantically sounds a bit more appropriate.